### DISCUSSION TITLE
Support for {.css-class} notation

### DISCUSSION THREAD

**Discussion by alkisg** [facebook/docusaurus#6023] [Q&A] [answered]
Hi, is there any method or extension that would allow docusaurus to support adding CSS classes to content, for example `![alt](img.png){.right-aligned}` or `# Header1 {.green-colored}` ?

Such syntax is supported by [many](https://michelf.ca/projects/php-markdown/extra/#spe-attr) [markdown](https://kramdown.gettalong.org/syntax.html#inline-attribute-lists) [libraries](https://www.npmjs.com/package/markdown-it-attrs) and we already use it in our content, and we'd like to be able to switch to docusaurus without modifying our existing markdown files...

**Comment by Josh-Cena** [comment, marked as answer]
Hi, for now there isn't one in Docusaurus. You can always build your own Remark plugin to do this: https://docusaurus.io/docs/markdown-features/plugins#creating-new-rehyperemark-plugins

We are aware of a potential solution: https://github.com/remarkjs/remark-directive But that's for MDX v2 while we are on MDX v1. We are soon planning for a version bump (which requires us to migrate our codebase from CJS to ESM), after which you should be able to enjoy this feature out-of-the-box.

**Comment by alkisg** [comment]
Thank you very much!
